### PR TITLE
NUTCH-2264 Check Forbidden API's at Build

### PR DESCRIPTION
**Pull request by kamaci**
Forbidden APIs is checked at ant build.

### CODE REVIEW

**Comment by kamaci** [comment]
This is an initial PR due to I get that error when I run ant check-forbidden-apis:

```
BUILD FAILED
/home/furkan/projects/gsoc2016/nutch/build.xml:1044: Check for forbidden API calls failed: java.lang.ClassNotFoundException: org.apache.avro.util.Utf8
        at de.thetaphi.forbiddenapis.Checker.getClassFromClassLoader(Checker.java:264)
        at de.thetaphi.forbiddenapis.Checker.lookupRelatedClass(Checker.java:277)
        at de.thetaphi.forbiddenapis.ClassScanner.checkType(ClassScanner.java:178)
        at de.thetaphi.forbiddenapis.ClassScanner.checkDescriptor(ClassScanner.java:218)
        at de.thetaphi.forbiddenapis.ClassScanner$1.<init>(ClassScanner.java:284)
        at de.thetaphi.forbiddenapis.ClassScanner.visitField(ClassScanner.java:279)
        at de.thetaphi.forbiddenapis.asm.ClassReader.a(Unknown Source)
        at de.thetaphi.forbiddenapis.asm.ClassReader.accept(Unknown Source)
        at de.thetaphi.forbiddenapis.asm.ClassReader.accept(Unknown Source)
        at de.thetaphi.forbiddenapis.Checker.checkClass(Checker.java:528)
        at de.thetaphi.forbiddenapis.Checker.run(Checker.java:545)
        at de.thetaphi.forbiddenapis.ant.AntTask.execute(AntTask.java:189)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
        at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.apache.tools.ant.Task.perform(Task.java:348)
        at org.apache.tools.ant.Target.execute(Target.java:435)
        at org.apache.tools.ant.Target.performTasks(Target.java:456)
        at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
        at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
        at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
        at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
        at org.apache.tools.ant.Main.runBuild(Main.java:851)
        at org.apache.tools.ant.Main.startAnt(Main.java:235)
        at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
        at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

File: build.xml [hunk 1]
```diff
@@ -1035,4 +1039,11 @@
       </classpath>
     </eclipse>
   </target>
+
+  <target name="check-forbidden-apis" depends="compile">
+    <fa:forbiddenapis internalRuntimeForbidden="true" classpathref="classpath" dir="${build.dir}">
+      <bundledsignatures name="jdk-unsafe-${javac.version}"/>
```
Comment (lewismc) [inline build.xml]: Are there no other tests which we could do here?

Comment (kamaci) [inline build.xml]: Here is a list for what we can do: [(https://github.com/policeman-tools/forbidden-apis/wiki/BundledSignatures)] On the other hand we can write some custom checks as like: [https://github.com/apache/lucene-solr/blob/master/lucene/tools/forbiddenApis/base.txt]

Comment (lewismc) [inline build.xml]: yes I think we should make full use of the plugin e.g. add more checks from the available options. If you wish to split these in to separate items and merge them gradually then I would be happy to merge. Please say what you want to do.

Comment (kamaci) [inline build.xml]: I've prepared much more complicated PR for this. However I got the same error. So, I wanted to solve it and then improve the PR. Do you have any idea about the error?

Comment (lewismc) [inline build.xml]: What error?

Comment (kamaci) [inline build.xml]: @lewismc I wrote it at my second comment for this PR:

`BUILD FAILED
/home/furkan/projects/gsoc2016/nutch/build.xml:1044: Check for forbidden API calls failed: java.lang.ClassNotFoundException: org.apache.avro.util.Utf8
        at de.thetaphi.forbiddenapis.Checker.getClassFromClassLoader(Checker.java:264)
        at de.thetaphi.forbiddenapis.Checker.lookupRelatedClass(Checker.java:277)
        at de.thetaphi.forbiddenapis.ClassScanner.checkType(ClassScanner.java:178)
        at de.thetaphi.forbiddenapis.ClassScanner.checkDescriptor(ClassScanner.java:218)
        at de.thetaphi.forbiddenapis.ClassScanner$1.<init>(ClassScanner.java:284)
        at de.thetaphi.forbiddenapis.ClassScanner.visitField(ClassScanner.java:279)
        at de.thetaphi.forbiddenapis.asm.ClassReader.a(Unknown Source)
        at de.thetaphi.forbiddenapis.asm.ClassReader.accept(Unknown Source)
        at de.thetaphi.forbiddenapis.asm.ClassReader.accept(Unknown Source)
        at de.thetaphi.forbiddenapis.Checker.checkClass(Checker.java:528)
        at de.thetaphi.forbiddenapis.Checker.run(Checker.java:545)
        at de.thetaphi.forbiddenapis.ant.AntTask.execute(AntTask.java:189)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
        at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.apache.tools.ant.Task.perform(Task.java:348)
        at org.apache.tools.ant.Target.execute(Target.java:435)
        at org.apache.tools.ant.Target.performTasks(Target.java:456)
        at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
        at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
        at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
        at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
        at org.apache.tools.ant.Main.runBuild(Main.java:851)
        at org.apache.tools.ant.Main.startAnt(Main.java:235)
        at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
        at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)`

**Comment by lewismc** [comment]
@kamaci Have you tried again recently?

**Comment by lewismc** [comment]
@kamaci PING

**Comment by lewismc** [comment]
Please close this PR off as it has been superceeded by #148 
Thanks @kamaci

**Comment by kamaci** [comment]
Closed due to has been superceeded by #148

**Pull request closed without merging**